File: app/src/main/java/no/joharei/flixr/photos/PhotosApi.java
package no.joharei.flixr.photos;

import javax.inject.Inject;

import no.joharei.flixr.api.FlickrApiContainer;
import no.joharei.flixr.api.models.PhotosPhotoset;
import rx.Observable;

public class PhotosApi {
    private FlickrApiContainer flickrApiContainer;

    @Inject
    PhotosApi(FlickrApiContainer flickrApiContainer) {
        this.flickrApiContainer = flickrApiContainer;
    }

    Observable<PhotosPhotoset> getPhotos(long photosetId, String userId) {
        return flickrApiContainer.getPhotos(photosetId, userId);
    }
}

File: app/src/main/java/no/joharei/flixr/photos/PhotosApi.kt
package no.joharei.flixr.photos

import io.reactivex.Observable
import no.joharei.flixr.api.FlickrApiContainer
import no.joharei.flixr.api.models.Photos
import no.joharei.flixr.tools.ObservableCache
import javax.inject.Inject

class PhotosApi
@Inject
internal constructor(private val flickrApiContainer: FlickrApiContainer, private val observableCache: ObservableCache) {
    internal fun getPhotos(photosetId: Long, userId: String?): Observable<Photos> {
        return observableCache.getCachedObservable(flickrApiContainer.getPhotos(photosetId, userId), Photos::class.java, "photoset:$photosetId", true, true)
    }

    internal fun clearCache() = observableCache.clearCache()
}
